//
// PROJECT:         Aspia Remote Desktop
// FILE:            proto/desktop_session.proto
// LICENSE:         See top-level directory
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "desktop_session_message.proto";
import "power_session_message.proto";
import "status.proto";

package aspia.proto.desktop;

message HostToClient
{
    Status status                              = 1;
    VideoPacket video_packet                   = 2;
    AudioPacket audio_packet                   = 3;
    CursorShape cursor_shape                   = 4;
    ClipboardEvent clipboard_event             = 5;
    DesktopSessionConfigRequest config_request = 6;
}

message ClientToHost
{
    PointerEvent pointer_event     = 1;
    KeyEvent key_event             = 2;
    ClipboardEvent clipboard_event = 3;
    PowerEvent power_event         = 4;
    DesktopSessionConfig config    = 5;
}
